cli/src/wallet.rs

```diff
@@ -911,6 +911,11 @@ pub fn process_transfer(
         None
     };
 
+    let compute_unit_limit = if nonce_account.is_some() {
+        ComputeUnitLimit::Default
+    } else {
+        ComputeUnitLimit::Simulated
+    };
```

Comment:
Why don't you simulate nonce transactions?

Reply:
(Sorry for the late response, I was out for a bit)

We took this approach with the token CLI in https://github.com/solana-labs/solana-program-library/pull/6550 to handle offline signers. Since offline signers can't access the network, we can't simulate the transaction before signing, so they'll need to use the default (or eventually static) compute unit limit.

When broadcasting the signed transaction, we need to create exactly the same transaction as the offline signers, so we use the presence of a nonce account as a guess that there are offline signers, since we can't tell from a `Signer` if it's a pre-signer.

While offline signers and nonces aren't directly related, I couldn't come up with another easy way of knowing if a transaction has pre-signers. Suggestions are very welcome!

Reply:
Yeah pretty much, the situation is more specifically if *any* of the signers was a pre-signer, in which case there's a high probability that it was signed offline.

The main reason that I resisted that approach is that the `--signer` or `--blockhash` args are parsed by helpers like `signer_from_path` and `BlockhashQuery`, and the top-level normally never references them directly. But you're right, those are probably clearer than just the presence of a nonce account.

If we do go with that approach, we already have the `BlockhashQuery` and can disable the simulation if it's `None` or `FeeCalculator`. For the pre-signers, we need to add another field to all of the command parsers, like `contains_presigners`, and just check if the `--signer` arg is present. What do you think?

Reply:
Hmm after thinking more I think the nonce account approach is a good enough signal for now. The offline signer / supplied blockhash signals are technically not perfect either. Makes me feel all of that could be cleaned up in a separate change but for now this approach is fine.